kogito-codegen-modules/kogito-codegen-predictions/src/test/java/org/kie/kogito/codegen/prediction/PredictionCodegenUtilsTest.java

```diff
@@ -0,0 +1,135 @@
+        final List<OutputField> outputFields = getRandomOutputFields();
+        outputFields.add(getRandomOutputField(miningFields.get(miningFields.size() - 1).getName()));
+        kiePMMLModel = getKiePMMLModelInternal(miningFields, outputFields);
+    }
+
+    @ParameterizedTest
+    @MethodSource("org.kie.kogito.codegen.api.utils.KogitoContextTestUtils#contextBuilders")
+    void checkModel(KogitoBuildContext.Builder contextBuilder) {
+        KiePMMLModel hasSourcesMap = new KiePMMLModelWithSources("fileName", "modelName", "kmodulePackageName",
+                Collections.emptyList(), Collections.emptyList(),
+                Collections.emptyList(), Collections.emptyMap(),
+                true);
+        IllegalStateException thrown = assertThrows(
+                IllegalStateException.class,
+                () -> PredictionCodegenUtils.checkModel(hasSourcesMap),
+                "Expected checkModel() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains("Unexpected HasSourcesMap instance"));
+
+        KiePMMLModel noModelName = getKiePMMLModelInternal("fileName", null, Collections.emptyList(),
+                Collections.emptyList());
+        thrown = assertThrows(
+                IllegalStateException.class,
+                () -> PredictionCodegenUtils.checkModel(noModelName),
+                "Expected checkModel() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains("Model name should not be empty"));
+
+        KiePMMLModel emptyModelName = getKiePMMLModelInternal("fileName", "", Collections.emptyList(),
+                Collections.emptyList());
+        thrown = assertThrows(
+                IllegalStateException.class,
+                () -> PredictionCodegenUtils.checkModel(emptyModelName),
+                "Expected checkModel() to throw, but it didn't");
+        assertTrue(thrown.getMessage().contains("Model name should not be empty"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("org.kie.kogito.codegen.api.utils.KogitoContextTestUtils#contextBuilders")
+    void generateModelBaseFiles(KogitoBuildContext.Builder contextBuilder) {
+        final Collection<GeneratedFile> files = new ArrayList<>();
+        String modelPath = "/model/path/PmmlModel.pmml";
+        Map<String, byte[]> compiledClasses = new HashMap<>();
+        IntStream.range(0, 3).forEach(i -> compiledClasses.put("generated.package.Class" + i, new byte[0]));
+        PMMLResource resource = new PMMLResource(Collections.singletonList(kiePMMLModel),
+                Path.of(modelPath),
+                modelPath,
+                compiledClasses,
+                Collections.emptyMap());
+        PredictionCodegenUtils.generateModelBaseFiles(files, resource);
+        assertThat(files).hasSize(compiledClasses.size());
+        compiledClasses.keySet().forEach(fullClassName -> {
+            assertThat(files.stream().anyMatch(file -> file.path().toString().equals(fullClassName))).isTrue();
```

Comment:
Simplify the condition like this:

```suggestion
            assertThat(files).extracting(file -> file.path().toString()).contains(fullClassName);
```

Reply:
@pibizza 
done